x-pack/plugins/fleet/server/services/artifacts/artifacts.ts

```diff
@@ -154,24 +157,29 @@ export const bulkCreateArtifacts = async (
         refresh,
       })
     );
+
     // Track errors of the bulk create action
     if (res.errors) {
       nonConflictErrors.push(
         ...res.items.reduce<Error[]>((acc, item) => {
-          if (item.create?.status !== 409) {
-            acc.push(new Error(item.create?.error?.reason));
+          // 409's (conflict - record already exists) are ignored since the artifact already exists
+          if (item.create && item.create.status !== 409) {
+            acc.push(
+              new Error(
+                `Create of artifact id [${item.create._id}] returned: result [${
+                  item.create.result
+                }], status [${item.create.status}], reason [${JSON.stringify(
+                  item.create?.error || ''
+                )}]`
+              )
+            );
           }
           return acc;
         }, [])
       );
     }
   }
 
-  // If any non conflict error, it returns only the errors
```

Comment:
Removing this, are we returning `artifacts` in `{ artifacts }` response that contain errors?

Reply:
Yes, exactly. `bulk*` type of methods should always return both items that were successfully created and any errors that were encountered. What was happening here on our side was that if artifacts were actually created, but (for example) one generated an error, then we would never update our Manifest with the ones that were created - which would then lead to them being orphan artifacts.

Reply:
Got it, should we also update then the Manifest Manager (`pushArtifacrts()`) side, so we handle those that contain errors and only add to the manifest those that have been correctly created?